api/machinemanager/machinemanager.go

```diff
@@ -158,14 +158,17 @@ func (client *Client) UpgradeSeriesPrepare(machineName, series string, force boo
 		Force:  force,
 	}
 	result := params.ErrorResult{}
-	err := client.facade.FacadeCall("UpgradeSeriesPrepare", args, &result)
-	if err != nil {
+	if err := client.facade.FacadeCall("UpgradeSeriesPrepare", args, &result); err != nil {
 		return errors.Trace(err)
 	}
-	if result.Error != nil {
-		return result.Error
-	}
 
+	err := result.Error
+	if err != nil {
+		if params.IsCodeAlreadyExists(err) {
```

Comment:
It might be better to use `IsUpgradeInProgress` rather than `IsCodeAlreadyExists` given that the call `UpgradeSeriesPrepare` and not which does reveal any kind of creation which would warrant and existence error.

Reply:
I am a little hesitant to dilute the meaning of the upgrade-in-progress error, as it is only used for model upgrades currently.

I will add some appropriate comments. It can always be changed if required.